anjani/plugins/welcome.py

```diff
@@ -109,26 +148,48 @@ async def _member_join(self, message: Message, reply_to: int) -> None:
                     )
 
                     if button:
-                        button = util.tg.build_button(button)
-                        msg = await self.bot.client.send_message(
-                            chat.id,
-                            formatted_text,
-                            reply_to_message_id=reply_to,
-                            reply_markup=button,
-                        )
+                        button = build_button(button)
                     else:
-                        msg = await self.bot.client.send_message(
-                            chat.id,
-                            formatted_text,
-                            reply_to_message_id=reply_to,
+                        button = None
+
+                    try:
+                        if msg_type in {Types.TEXT, Types.BUTTON_TEXT}:
+                            msg = await self.SEND[msg_type](
+                                message.chat.id,
+                                formatted_text,
+                                reply_to_message_id=reply_to,
+                                reply_markup=button,
+                                disable_web_page_preview=True,
+                            )
+                        elif msg_type in {Types.STICKER, Types.ANIMATION}:
+                            msg = await self.SEND[msg_type](
+                                message.chat.id,
+                                file_id,
+                                reply_to_message_id=reply_to,
+                            )
+                        else:
+                            msg = await self.SEND[msg_type](
+                                message.chat.id,
+                                file_id,
+                                caption=formatted_text,
+                                reply_to_message_id=reply_to,
+                                reply_markup=button,
+                            )
+                    except MediaEmpty:
+                        await self.bot.client.send_message(
+                            message.chat.id,
+                            await self.text(message.chat.id, "welcome-message-expired"),
                         )
-
-                    previous = await self.previous_welcome(chat.id, msg.id)
-                    if previous:
-                        try:
-                            await self.bot.client.delete_messages(chat.id, previous)
-                        except MessageDeleteForbidden:
-                            pass
+                    except MessageEmpty:
+                        self.log.warning("Welcome message empty on %s.", message.chat.id)
+
+                    if msg:
```

Comment:
`msg` here are possibly unbound when it run the exception block

Reply:
What do you prefer? should I initialize the msg as None or should return a msg variable on exception.

Reply:
better to declare it as `None`